```
$ bin/test sympy/combinatorics
______________________________________________________________________________________________________________________________________________________________________
_______________________________________________________ sympy/combinatorics/tests/test_subsets.py:test_subset ________________________________________________________
Traceback (most recent call last):
  File "/mnt/c/Users/paul/Documents/PythonScripts/Libs/SymPy/sympy/sympy/combinatorics/tests/test_subsets.py", line 7, in test_subset
    assert a.next_binary() == Subset(['b'], ['a', 'b', 'c', 'd'])
AssertionError

======================================= tests finished: 158 passed, 1 failed, 3 skipped, 3 expected to fail, in 23.22 seconds ========================================
DO *NOT* COMMIT!
```

--- a/sympy/combinatorics/subsets.py
+++ b/sympy/combinatorics/subsets.py
@@ -1,10 +1,9 @@
 from itertools import combinations
 
 from sympy.combinatorics.graycode import GrayCode
-from sympy.core import Atom
 
 
-class Subset(Atom):
+class Subset():
     """
     Represents a basic subset object.
 
@@ -59,10 +58,25 @@ class Subset(Atom):
             if elem not in superset:
                 raise ValueError('The superset provided is invalid as it does '
                                  'not contain the element {}'.format(elem))
-        obj = Atom.__new__(cls)
+        obj = object.__new__(cls)
         obj._subset = subset
         obj._superset = superset
         return obj
+
+    def __eq__(self, other):
+        """Return a boolean indicating whether a == b on the basis of
+        whether both objects are of the class Subset and if the values 
+        of the subset and superset attributes are the same.
+        """
+        if not isinstance(self, Subset):
+            return False
+        if not isinstance(other, Subset):
+            return False
+        if not self.superset == other.superset:
+            return False
+        if not self.subset == other.subset:
+            return False
+        return True
 
     def iterate_binary(self, k):
         """

--- a/sympy/combinatorics/subsets.py
+++ b/sympy/combinatorics/subsets.py
@@ -65,18 +65,12 @@ class Subset():
 
     def __eq__(self, other):
         """Return a boolean indicating whether a == b on the basis of
-        whether both objects are of the class Subset and if the values 
+        whether both objects are of the class Subset and if the values
         of the subset and superset attributes are the same.
         """
-        if not isinstance(self, Subset):
-            return False
         if not isinstance(other, Subset):
-            return False
-        if not self.superset == other.superset:
-            return False
-        if not self.subset == other.subset:
-            return False
-        return True
+            return NotImplemented
+        return self.subset == other.subset and self.superset == other.superset
 
     def iterate_binary(self, k):
         """

--- a/sympy/combinatorics/tests/test_subsets.py
+++ b/sympy/combinatorics/tests/test_subsets.py
@@ -54,6 +54,8 @@ def test_subset():
     raises(ValueError, lambda: Subset(['a'], ['b', 'c']))
     raises(ValueError, lambda: Subset.subset_from_bitlist(['a', 'b'], '010'))
 
+    assert Subset(['a'], ['a', 'b']) != Subset(['b'], ['a', 'b'])
+    assert Subset(['a'], ['a', 'b']) != Subset(['a'], ['a', 'c'])
 
 def test_ksubsets():
     assert list(ksubsets([1, 2, 3], 2)) == [(1, 2), (1, 3), (2, 3)]

--- a/sympy/core/tests/test_args.py
+++ b/sympy/core/tests/test_args.py
@@ -500,12 +500,6 @@ def test_sympy__combinatorics__graycode__GrayCode():
     assert _test_args(GrayCode(3, rank=1))
 
 
-def test_sympy__combinatorics__subsets__Subset():
-    from sympy.combinatorics.subsets import Subset
-    assert _test_args(Subset([0, 1], [0, 1, 2, 3]))
-    assert _test_args(Subset(['c', 'd'], ['a', 'b', 'c', 'd']))
-
-
 def test_sympy__combinatorics__permutations__Permutation():
     from sympy.combinatorics.permutations import Permutation
     assert _test_args(Permutation([0, 1, 2, 3]))

